### PR TITLE
[adsp] rework to new system who already on PVR

### PR DESCRIPTION
**Pull request by AlwinEsch**
This add also the changes already present on PVR to the ADSP system.

Has found during this a fault on PVR system. With the `m_PVRManager.reset(new PVR::CPVRManager());`called on `Init2` comes a fault during settings creation where PVR pointer was null!

@FernetMenta can you recheck it, the fix for PVR is on last commit here.
@AchimTuran can you review the rest.

### CODE REVIEW

**Comment by FernetMenta** [comment]
@AlwinEsch I would fix it differently. Settings is a service too and should not depend on PVRManager which is a client to this service. CSettingsManager::RegisterCallback is a public method. Instead of registering pvrmanager in CSettings::InitializeISettingCallbacks, pvrmanager should do this by itself when ready.

**Comment by Montellese** [comment]
@FernetMenta: that was my final goal as well but during the whole settings refactor it was easier to control and to test by having all of the callback registration being done in the same place. But for all `ISettingCallbacks` implementations this could be moved to their own code so that it is only done when necessary and when the implementation is ready.

**Comment by FernetMenta** [comment]
for reference: https://github.com/xbmc/xbmc/pull/9484
the interface was already in place

**Comment by AlwinEsch** [comment]
Can someone explain me why travis fails with this?
```
In file included from /home/travis/build/xbmc/xbmc/xbmc/cores/AudioEngine/Utils/AEAudioFormat.h:23:0,
                 from /home/travis/build/xbmc/xbmc/xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.h:23,
                 from /home/travis/build/xbmc/xbmc/xbmc/ServiceManager.h:25,
                 from /home/travis/build/xbmc/xbmc/xbmc/Application.h:30,
                 from /home/travis/build/xbmc/xbmc/xbmc/music/dialogs/GUIDialogMusicOSD.cpp:29:
/home/travis/build/xbmc/xbmc/xbmc/cores/AudioEngine/Utils/AEStreamInfo.h:30:27: fatal error: libavutil/crc.h: No such file or directory
 #include "libavutil/crc.h"
                           ^
compilation terminated.
make[2]: *** [build/music/dialogs/CMakeFiles/music_dialogs.dir/GUIDialogMusicOSD.cpp.o] Error 1
make[1]: *** [build/music/dialogs/CMakeFiles/music_dialogs.dir/all] Error 2
```

Here at home it works fine with cmake and have not changed something about them.

**Comment by fetzerch** [comment]
@AlwinEsch: Add `ffmpeg` to the target dependencies in `xbmc/music/dialogs/CMakeLists.txt`.
(You can see that the compilation of `GUIDialogMusicOSD.cpp` happens in parallel to ffmpeg so the include is not yet available).

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

File: system/settings/settings.xml [hunk 1]
```diff
@@ -2561,40 +2561,13 @@
         </setting>
       </group>
       <group id="2">
-        <setting id="audiooutput.dspaddonsenabled" type="boolean" label="36441" help="36438">
```
Comment (AchimTuran) [inline system/settings/settings.xml]: @FernetMenta @AlwinEsch 
This is really the way we should go. Automatically enable AudioDSP if the first add-on is enable, but my opinion is that it is to early to completly remove a button to enable/disable AudioDSP. Currently we have some issues with and it doesn't work properly on all platforms. I'm not a blocker, but I like to give the users the possibility to disable it completly until we have good support for all platforms.
According to @AlwinEsch we have disable all add-ons manually to disable AudioDSP completly.

What do you think about it?

File: xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp [hunk 2]
```diff
@@ -75,36 +74,29 @@ CActiveAEDSP::CActiveAEDSP()
...
-/*! @name initialization and configuration methods */
-//@{
-class CActiveAEDSPStartJob : public CJob
+void CActiveAEDSP::Init(void)
 {
-public:
-  CActiveAEDSPStartJob() {}
-  ~CActiveAEDSPStartJob(void) {}
+  std::set<std::string> settingSet;
+  settingSet.insert(CSettings::SETTING_AUDIOOUTPUT_DSPSETTINGS);
```
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp]: To be consistent would it be possible to rename `SETTING_AUDIOOUTPUT_DSPSETTINGS` to `SETTING_AUDIOOUTPUT_ADSPSETTINGS`?

**Comment by AchimTuran** [comment]
@AlwinEsch 
I have no further suggestions. ;-)

File: system/settings/settings.xml (hunk 1 above)
Comment (FernetMenta) [inline system/settings/settings.xml]: just disable the addon. why would someone want to have an addon enable and use a different switch to disable it again?

Comment (FernetMenta) [inline system/settings/settings.xml]: I agree.

**Comment by AlwinEsch** [comment]
Have brought back the "Enable DSP" option.

jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**